src/stratigraphy/elevation/elevation_extraction.py

```diff
@@ -0,0 +1,268 @@
+
+    The ElevationExtractor class is used to extract elevation data from text. The class is designed
+    to be used in combination with the StratigraphyExtractor class to extract elevation data from
+    stratigraphic descriptions.
+
+    The class provides methods to extract elevation data from text. The main method is the extract_elevation
+    method which extracts elevation data from a list of text lines. The class also provides methods to find
+    the location of a coordinate key in a string of text.
+    """
+
+    def __init__(self, document: fitz.Document):
+        """Initializes the CoordinateExtractor object.
+
+        Args:
+            document (fitz.Document): A PDF document.
+        """
+        self.doc = document
+        self.elevation_keys = read_params("matching_params.yml")["elevation_keys"]
+
+    def find_elevation_key(self, lines: list[TextLine], allowed_errors: int = 3) -> TextLine | None:  # noqa: E501
+        """Finds the location of a coordinate key in a string of text.
+
+        This is useful to reduce the text within which the coordinates are searched. If the text is too large
+        false positive (found coordinates that are no coordinates) are more likely.
+
+        The function allows for a certain number of errors in the key. Errors are defined as insertions, deletions
+        or substitutions of characters (i.e. Levenshtein distance). For more information of how errors are defined see
+        https://github.com/mrabarnett/mrab-regex?tab=readme-ov-file#approximate-fuzzy-matching-hg-issue-12-hg-issue-41-hg-issue-109.
+
+
+        Args:
+            lines (list[TextLine]): Arbitrary text lines to search in.
+            allowed_errors (int, optional): The maximum number of errors (Levenshtein distance) to consider a key
+                                            contained in text. Defaults to 3 (guestimation; no optimisation done yet).
+
+        Returns:
+            TextLine | None: The line of the coordinate key found in the text.
+        """
+        matches = []
+        for key in self.elevation_keys:
+            if len(key) < 5:
+                # if the key is very short, do an exact match
+                pattern = regex.compile(r"\b" + key + r"\b", flags=regex.IGNORECASE)
+            else:
+                pattern = regex.compile(r"\b" + key + "{e<" + str(allowed_errors) + r"}\b", flags=regex.IGNORECASE)
+            for line in lines:
+                match = pattern.search(line.text)
+                if match:
+                    matches.append((line, sum(match.fuzzy_counts)))
+
+        # if no match was found, return None
+        if len(matches) == 0:
+            return None
+
+        # Remove duplicates
+        matches = list(dict.fromkeys(matches))
+
+        # Sort the matches by their error counts (ascending order)
+        matches.sort(key=lambda x: x[1])
+
+        # Return the top three matches (lines only)
+        return [match[0] for match in matches[:5]]
```

Comment:
Can we remove the selection of the "best n matched" and improve the deduplication, analogous to https://github.com/swisstopo/swissgeol-boreholes-dataextraction/pull/72/commits/ee46a8267fdff082660ad5f7a0a107fe70e51ea9 ?

Reply:
Yes I will do so.